controllers/managedocs_controller.go

```diff
@@ -678,8 +678,13 @@ func (r *ManagedOCSReconciler) reconcileAlertmanager() error {
 		}
 
 		desired := templates.AlertmanagerTemplate.DeepCopy()
-		r.alertmanager.ObjectMeta.Labels = map[string]string{monLabelKey: monLabelValue}
+		desired.Spec.AlertmanagerConfigSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				monLabelKey: monLabelValue,
+			},
+		}
```

Comment:
I think this can be moved to the alertmanager.go template. Since we are adding the selectors for other resources in the templates. 
`
AlertmanagerConfigSelector: &resourceSelector`

Also , please remove the ConfigSecret from the alertmanager.go template, as we are not using it anymore

Reply:
@kesavanvt  I actually disagree here, we have constants that hold the values of the label key and label value. We should use them and not repeat the string values somewhere else (even if it is a template). I will suggest doing one of the following:
Either allow the templates to have access to the const and use them in the template,
or remove the selector from all templates and configure everything in the reconciler where we have access to the consts